### PR TITLE
alt5 method-based syntax updates to all chapters

### LINKED ISSUE #167: Update to Altair 5
Altair 5 was recently released and it fixes some issues from before (like more reliable saving and working with large datasets, https://altair-viz.github.io/releases/changes.html), as well as introduces a more convenient (optional) syntax for parameter options. I'm happy to make the changes if we are OK updating the package on our service for the fall. This should be done after #160 is merged imo.

### PR DESCRIPTION
**Pull request by joelostblom**
closes #167

### CODE REVIEW

File: source/inference.md [hunk 1]
```diff
@@ -645,15 +645,23 @@ glue(
...
+        sample_distribution.encode(
+            x=alt.X("price")
+                .bin(extent=[0, 660], maxbins=40)
+                .title("Price per night (Canadian dollars)")
+        ).properties(title="Sample (n = 40)").properties(height=150),
+        sampling_distribution.encode(
+            x=alt.X("sample_mean")
+                .bin(extent=[0, 660], maxbins=40)
+                .title("Price per night (Canadian dollars)")
+        ).properties(
```
Comment (joelostblom) [inline source/inference.md:664]: This fix changes the appearance from uneven bins:

image

to even bins:

image

which is easier to compare (and more similar to the R version):

image

**Review by trevorcampbell — CHANGES_REQUESTED** [review]
Lots of great changes here! Some comments/fixes needed:

- Multiple figures in classification 1 aren't rendering any more, and broken figure references
- some bars in inference are too thick now (just one example here but there are many)

e.g.

image

vs

image

- Some figure aspect ratios have changed (for the worse)
  - but some figures also seem to fit my browser window better now. not sure if this was an intentional change or if it's just a new default in altair / jupyter-book or something
  
- syntax highlighting seems to have changed, FYI, compared to the current public website. But I'm fine with it, it looks nice to me -- probably not related to anything you did, just the new docker image

image

vs

image

File: source/inference.md (hunk 1 above)
Comment (trevorcampbell) [reply source/inference.md:664]: nice catch!

**Comment by joelostblom** [comment]
Thanks @trevorcampbell !

> Multiple figures in classification 1 aren't rendering any more, and broken figure references

Good catch, fixed I think. I only see warnings about missing citation keys when I build the HTML now, but I believe those were there previously:

```
/home/jovyan/source/viz.md:2021: WARNING: duplicate citation for key "mckinney2012python"
/home/jovyan/source/viz.md:2021: WARNING: duplicate citation for key "timbers2020canlang"
/home/jovyan/source/wrangling.md:1822: WARNING: duplicate citation for key "mckinney2012python"
/home/jovyan/source/classification2.md:1995: WARNING: duplicate citation for key "james2013introduction"
/home/jovyan/source/clustering.md:794: WARNING: duplicate citation for key "james2013introduction"
/home/jovyan/source/inference.md:1347: WARNING: duplicate citation for key "moderndive"
/home/jovyan/source/classification1.md:1889: WARNING: duplicate citation for key "knncover"
/home/jovyan/source/classification1.md:1889: WARNING: duplicate citation for key "knnfix"
/home/jovyan/source/classification1.md:1889: WARNING: duplicate citation for key "streetbreastcancer"
```


> some bars in inference are too thick now (just one example here but there are many)

Fixed by adding back `maxbins=30` for the charts where it was removed

> Some figure aspect ratios have changed (for the worse) but some figures also seem to fit my browser window better now. not sure if this was an intentional change or if it's just a new default in altair / jupyter-book or something.

In Altair 5 we changed the default from a 4:3 ratio to a square 3:3 ratio to avoid distortion in scatter plots and align with how things are done in the underlying Vega-Lite library. If there are specific figures where you think this is unhelpful we could set the width manually.

> syntax highlighting seems to have changed, FYI, compared to the current public website. But I'm fine with it, it looks nice to me -- probably not related to anything you did, just the new docker image

I noticed this as well and it looks OK to me too.

**Review by trevorcampbell — APPROVED** [review]
lgtm

**Pull request merged**